File: OffByOne/src/OffByOne.LanguageCore/Ast/Expressions/Binary/Base/BinaryExpression.cs
namespace OffByOne.LanguageCore.Ast.Expressions.Binary.Base
{
    public abstract class BinaryExpression : Expression
    {
        protected BinaryExpression(
            Expression leftExpression,
            Expression rightExpression)
        {
            this.LeftExpression = leftExpression;
            this.RightExpression = rightExpression;
        }

        public Expression LeftExpression { get; private set; }

        public Expression RightExpression { get; private set; }
    }
}

File: Hatt/taxform/src/main/java/org/uva/hatt/taxform/ast/nodes/expressions/BinaryExpression.java
package org.uva.hatt.taxform.ast.nodes.expressions;

public abstract class BinaryExpression extends Expression{

    private final Expression lhs;
    private final Expression rhs;

    protected BinaryExpression(int lineNumber, Expression lhs, Expression rhs) {
        super(lineNumber);
        this.lhs = lhs;
        this.rhs = rhs;
    }

    public Expression getLhs() {
        return lhs;
    }

    public Expression getRhs() {
        return rhs;
    }
}
